lib/src/contact/contact_list.dart

```diff
@@ -316,33 +316,53 @@ class _ContactListState extends State<ContactList> {
                   }),
               dismissal: SlidableDismissal(
                 child: SlidableDrawerDismissal(),
-                onDismissed: (actionType) {
-                  if (actionType == SlideActionType.primary) {
-                    _blockContact(contactId: contactId);
-                  } else {
-                    _deleteContact(contactId: contactId);
+                onWillDismiss: (actionType) async {
+                  if (actionType == SlideActionType.secondary) {
+                    await _slidableActionCalled(contactId: contactId, action: SlidableAction.delete);
+                  } else if (actionType == SlideActionType.primary) {
+                    await _slidableActionCalled(contactId: contactId, action: SlidableAction.block);
+                    return true;
                   }
+                  return false;
                 },
               ),
-              child: ContactItem(
-                  contactId: contactId,
-                  contactItemType: ContactItemType.edit,
-                  key: key));
+              child: ContactItem(contactId: contactId, contactItemType: ContactItemType.edit, key: key));
         });
   }
 
-  // Slide Actions
-
-  _blockContact({@required int contactId}) {
+  Future<bool> _slidableActionCalled({@required int contactId, @required SlidableAction action}) async {
     ContactChangeBloc bloc = ContactChangeBloc();
-    bloc.add(BlockContact(contactId: contactId));
-    bloc.close();
-  }
+    switch (action) {
+      case SlidableAction.block:
+        bloc.add(BlockContact(contactId: contactId));
+        break;
+      case SlidableAction.delete:
+        bloc.add(DeleteContact(id: contactId));
+        break;
+    }
 
-  _deleteContact({@required int contactId}) {
-    ContactChangeBloc bloc = ContactChangeBloc();
-    bloc.add(DeleteContact(id: contactId));
-    bloc.close();
+    await for (ContactChangeState state in bloc) {
```

Comment:
Just for the curious: Could that be possible too, for pre-filtering requested `contactId`:
```
await for (ContactChangeState state in bloc.where((b) => b.contactId == contactId).toList())
```
or to be a bit more readable:
```
ContactChangeBloc filteredBloc = bloc.where((b) => b.contactId == contactId).toList();
await for (ContactChangeState state in filteredBloc)
```
Going this way we can get rid of the two `&& state.contactId == contactId` statements below.  
Would that be possible? 🤔

But I guess the `toList()` is the blocker here...

Reply:
I think that won't work ;) but we could wrap the whole inner for loop part in one contactId == statement to avoid duplication.